demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -267,33 +274,61 @@ static uint32_t prvGetTimeMs( void );
 
 /*-----------------------------------------------------------*/
 
-static int32_t prvGenerateRandomNumber()
+static BaseType_t prvBackoffForRetry( BackoffAlgorithmContext_t * pxRetryParams )
 {
-    uint32_t ulRandomNum;
+    BaseType_t xReturnStatus = pdFAIL;
+    uint16_t usNextRetryBackOff = 0U;
+    BackoffAlgorithmStatus_t xBackoffAlgStatus = BackoffAlgorithmSuccess;
+
+    /**
+     * To calculate the backoff period for the next retry attempt, we will
+     * generate a random number to provide to the backoffAlgorithm library.
+     *
+     * Note: The PKCS11 module is used to generate the random number as it allows access
+     * to a True Random Number Generator (TRNG) if the vendor platform supports it.
+     * It is recommended to use a random number generator seeded with a device-specific
+     * entropy source so that probability of collisions from devices in connection retries
+     * is mitigated.
```

Comment:
For the demos using this helper function, I think it is better to provide a comment in the demo file itself so that it is easily noticeable. The comment in this file may not be easy to notice from a demo using it.

Reply:
Similar comment for the demos using http utils also.

Reply:
This helper is not being used directly by the demos though. The demos only call the `EstablishMqttSession` helper function.
This detail may not be relevant to the reader at the call-site in the Shadow, Jobs and Defender demos...